managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -175,6 +177,8 @@ public class ManagedCursorImpl implements ManagedCursor {
     // Stat of the cursor z-node
     private volatile Stat cursorLedgerStat;
 
+    private volatile ManagedCursorInfo managedCursorInfo;
```

Comment:
Introducing this variable worries me a bit, because it means remembering to update this variable every update. otherwise, we may lose some data in managed cursor info.

Reply:
Yes, but it can reduce a get operation

Reply:
could we add the comments on here or `asyncUpdateCursorInfo` method to alter other contributors won't forget to update it? 
also, we need to ensure the updated order.
- managedCursorInfo first
- cursorLedgerStat last
if it is out of order may cause the update will use the new cursorLedgerStat value and old managedCursorInfo.

Because the concurrent problem is hard to find and debug. thanks!

Reply:
Also, I think one more get operation might be fine since this feature isn't used very often.
But when we add this variable it will affect other logic and introduce the risk of data inconsistency.

I think we need to reach a consensus here.
/cc @codelipenghui  @lhotari  @eolivelli @Jason918  WDYT?

Reply:
maybe I missing usage for PIP-195, I will check it tomorrow.

Reply:
> could we add the comments on here or asyncUpdateCursorInfo method to alter other contributors won't forget to update it?

I package `updateCursorLedgerStat` method to update cursorLedgerStat and managedCursorInfo at the same time and add some notes to alter other contributors don't update individually cursorLedgerStat.

Reply:
> Also, I think one more get operation might be fine since this feature isn't used very often.
> But when we add this variable it will affect other logic and introduce the risk of data inconsistency.

@mattisonchao 
This PR have this "get" at first. It's removed under Penghui's suggestion.
After check the usage of `cursorLedgerStat`, I think it's ok to drop the "get" op.

Reply:
Got it, thanks!